joblib/_parallel_backends.py

```diff
@@ -43,10 +43,13 @@ def supports_timeout(self):
 
     nesting_level = None
 
-    def __init__(self, nesting_level=None, inner_max_num_threads=None, **kwargs):
-        super().__init__(**kwargs)
+    def __init__(
+        self, nesting_level=None, inner_max_num_threads=None, **backend_kwargs
+    ):
```

Comment:
Instead of adding extra arguments for each backend, I am in favor of having a generic way of passing the argument and better documenting the available options.

Reply:
As we are refactoring the documentation on parallel backend in #1667 , I would say we can improve the backend documentation once these two PR are merged.